### ISSUE TITLE
failing : cmake .. -DCMAKE_BUILD_TYPE=Release -DW2L_CRITERION_BACKEND=CUDA . cannot find CharLMDecoder.cpp

### ISSUE THREAD

**Issue by megharangaswamy** [facebookresearch/wav2letter#299]
I am trying to do decoder stage and I ended up with segmentation fault. when I checked a few issues related to that, it was suggested to take the latest version of wav2letter++ because a  few new things were fixed in the latest code. But when I am making 

> cmake .. -DCMAKE_BUILD_TYPE=Release -DW2L_CRITERION_BACKEND=CUDA

 I am getting below error.
```
-- ArrayFire found (include: /opt/arrayfire/include, library: ArrayFire::afcuda)
-- Found glog    (include: /usr/local/include, library: /usr/local/lib/libglog.so)
-- GLOG found
-- Found gflags  (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libgflags.so)
-- GFLAGS found
-- OpenMP found
-- flashlight found (include:  lib: flashlight::flashlight )
-- flashlight built in distributed mode.
-- flashlight built with contrib features.
-- Checking for [mkl_gf_lp64 - mkl_gnu_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_gf_lp64: /opt/intel/mkl/lib/intel64/libmkl_gf_lp64.so
--   Library mkl_gnu_thread: /opt/intel/mkl/lib/intel64/libmkl_gnu_thread.so
--   Library mkl_core: /opt/intel/mkl/lib/intel64/libmkl_core.so
--   Library iomp5: not found
-- Checking for [mkl_gf_lp64 - mkl_intel_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_gf_lp64: /opt/intel/mkl/lib/intel64/libmkl_gf_lp64.so
--   Library mkl_intel_thread: /opt/intel/mkl/lib/intel64/libmkl_intel_thread.so
--   Library mkl_core: /opt/intel/mkl/lib/intel64/libmkl_core.so
--   Library iomp5: not found
-- Checking for [mkl_gf - mkl_gnu_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_gf: not found
-- Checking for [mkl_gf - mkl_intel_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_gf: not found
-- Checking for [mkl_intel_lp64 - mkl_gnu_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_intel_lp64: /opt/intel/mkl/lib/intel64/libmkl_intel_lp64.so
--   Library mkl_gnu_thread: /opt/intel/mkl/lib/intel64/libmkl_gnu_thread.so
--   Library mkl_core: /opt/intel/mkl/lib/intel64/libmkl_core.so
--   Library iomp5: not found
-- Checking for [mkl_intel_lp64 - mkl_intel_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_intel_lp64: /opt/intel/mkl/lib/intel64/libmkl_intel_lp64.so
--   Library mkl_intel_thread: /opt/intel/mkl/lib/intel64/libmkl_intel_thread.so
--   Library mkl_core: /opt/intel/mkl/lib/intel64/libmkl_core.so
--   Library iomp5: not found
-- Checking for [mkl_intel - mkl_gnu_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_intel: not found
-- Checking for [mkl_intel - mkl_intel_thread - mkl_core - iomp5 - pthread - m]
--   Library mkl_intel: not found
-- Checking for [mkl_gf_lp64 - mkl_gnu_thread - mkl_core - pthread - m]
--   Library mkl_gf_lp64: /opt/intel/mkl/lib/intel64/libmkl_gf_lp64.so
--   Library mkl_gnu_thread: /opt/intel/mkl/lib/intel64/libmkl_gnu_thread.so
--   Library mkl_core: /opt/intel/mkl/lib/intel64/libmkl_core.so
--   Library pthread: /usr/lib/x86_64-linux-gnu/libpthread.so
--   Library m: /usr/lib/x86_64-linux-gnu/libm.so
-- MKL library found
-- MKL found
-- FFTW found
-- Required SndFile dependency Ogg found.
-- Required SndFile dependency Vorbis found.
-- Required SndFile dependency VorbisEnc found.
-- Required SndFile dependency FLAC found.
-- Found libsndfile: (lib: /usr/lib/x86_64-linux-gnu/libsndfile.so include: /usr/include
-- libsndfile found.
-- Looking for KenLM
-- Using kenlm library found in /media/home/megha/5_wav2letter/kenlm/build/lib/libkenlm.a
-- Using kenlm utils library found in /media/home/megha/5_wav2letter/kenlm/build/lib/libkenlm.a
-- kenlm lm/model.hh found in /media/home/megha/5_wav2letter/kenlm/lm/model.hh
-- Found kenlm (include: /media/home/megha/5_wav2letter/kenlm, library: /media/home/megha/5_wav2letter/kenlm/build/lib/libkenlm.a;/media/home/megha/5_wav2letter/kenlm/build/lib/libkenlm_util.a)
-- kenlm found
-- LZMA found (library: /usr/lib/x86_64-linux-gnu/liblzma.so include: /usr/include)
-- BZip2 found (library: /usr/lib/x86_64-linux-gnu/libbz2.so include: /usr/include)
-- Z found (library: /usr/lib/x86_64-linux-gnu/libz.so include: /usr/include)
-- Adding warpctc:
-- warpctc: cuda found TRUE
-- warpctc: using CUDA 9.0 or above
-- warpctc: Building shared library with GPU support
-- CUDA found (library: /usr/local/cuda/lib64/libcudart_static.a;-lpthread;dl;/usr/lib/x86_64-linux-gnu/librt.so include: /usr/local/cuda/include)
-- CUDA_architecture_build_targets detected:  6.1
-- Configuring done
CMake Error at CMakeLists.txt:134 (add_executable):
  Cannot find source file:

    /media/home/megha/5_wav2letter/WAV_2_LETTER/wav2letter/src/decoder/CharLMDecoder.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx


CMake Error at src/tests/CMakeLists.txt:9 (add_executable):
  Cannot find source file:

    /media/home/megha/5_wav2letter/WAV_2_LETTER/wav2letter/src/decoder/CharLMDecoder.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
Call Stack (most recent call first):
  src/tests/CMakeLists.txt:64 (build_test)


-- Generating done
-- Build files have been written to: /media/home/megha/5_wav2letter/WAV_2_LETTER/wav2letter/build
```
As the  error tells, I am not able to find any file called **CharLMDecoder.cpp** inside src/decoder. Kindly help me with this. I see a few new check-ins in the code related to CharLM. And I found
 WordLMDecoder.cpp inside decoder folder but not CharLMDecoder.cpp. I am not sure if I am thinking it right. Kindly help me to solve this.
@xuqiantong, can you confirm if I need to change anything here?

Thanks,
Megha

**Comment by jacobkahn** [comment]
@megharangaswamy — thanks for flagging. This is a bug. Someone will send a fix shortly.

**Comment by jacobkahn** [comment]
Fixed in b9fa15c46a487db6e4afce6c83d30ef76165a48d.
